app/controllers/members_controller.rb

```diff
@@ -62,6 +77,11 @@ def graph
     @link_data_str = edges.to_json.html_safe
   end
 
+  def check_email_confirmed!
+    @member = Member.find(params[:id])
+    redirect_to action: :index unless @member.email_confirmed
```

Comment:
No big deal, but try to make methods that answer questions with a question mark, like `email_confirmed?.